src/main/java/bowling/views/ResultView.java

```diff
@@ -0,0 +1,127 @@
+package bowling.views;
+
+import bowling.domain.*;
+import bowling.utils.StringUtils;
+
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+public class ResultView {
+    private final static String STRIKE = "X";
+    private final static String SPARE = "/";
+    private final static String GUTTER = "-";
+    private final static String SEPARATOR = "|";
+    private final static int GUTTER_PIN_COUNT = 0;
+
+    private final static int FIRST_ROUND = 0;
+    private final static int FINAL_ROUND = 10;
+    private final static int FIRST_STATE = 1;
+    private final static int FINAL_STATE = 3;
+
+    private final static int FIRST_INDEX = 0;
+    private final static int FRAME_WIDTH = 6;
+    private final static int FRAME_WIDTH_HALF = 2;
+
+    private final static String SCORE_TITLE = "| NAME |  01  |  02  |  03  |  04  |  05  |  06  |  07  |  08  |  09  |  10  |";
+    private final static String BLANK = "    ";
+    private final static String ENTER = System.lineSeparator();
+
+    private ResultView() {}
+
+    private static void showScoreBoardTitle() {
+        System.out.println(SCORE_TITLE);
+    }
+
+    private static String getStateOrName(String result) {
+        int rightPadding = (FRAME_WIDTH - result.length()) / FRAME_WIDTH_HALF;
+        int leftPadding = FRAME_WIDTH - result.length() - rightPadding;
+        return SEPARATOR +
+                StringUtils.generateBlank(leftPadding) +
+                result +
+                StringUtils.generateBlank(rightPadding);
+    }
+
+    private static String getBlankFrameStates() {
+        StringBuilder result = new StringBuilder();
+        for (int i = ResultView.FIRST_ROUND; i < FINAL_ROUND; i++) {
+            result.append(getStateOrName(BLANK));
+        }
+        return result.toString();
+    }
+
+    private static String getBlankOrFrameState(Frame frame) {
+        if (frame.hasResult()) {
+            return getFrameStates(frame.states(), frame.downPins());
+        }
+        return getStateOrName(BLANK);
+    }
+
+    private static String getFrameStates(List<State> states, DownPins downPins) {
+        if (states.size() == FIRST_STATE) {
+            return getFrameState(states.get(FIRST_INDEX), downPins);
+        }
+        String value = getFinalFrameStates(states, downPins);
+        return getStateOrName(value);
+    }
+
+    private static String getFinalFrameStates(List<State> states, DownPins downPins) {
+        return IntStream.range(FIRST_INDEX, states.size())
+                .mapToObj(i -> {
+                    if (State.STRIKE.equals(states.get(i))) {
+                        return STRIKE;
+                    }
+                    return getSpareOrGutter(downPins.get(i).count());
+                })
+                .collect(Collectors.joining(SEPARATOR));
```

Comment:
```suggestion
        return IntStream.range(FIRST_INDEX, states.size())
                .mapToObj(i ->  State.STRIKE.equals(states.get(i))
                        ? STRIKE
                        : getSpareOrGutter(downPins.get(i).count())
                ).collect(Collectors.joining(SEPARATOR));
```

함수형을 활용해 주신 점 매우 좋습니다 👍 
취향 차이이기는 하지만, map 안에 if 문이 있다면 삼항 연산자를 활용해보거나, 별도의 메서드로 분리해보시는 건 어떤가요?
취향차이이므로 꼭 수정해주시지 않아도 괜찮습니다.